mne/epochs.py

```diff
@@ -3089,5 +3089,5 @@ def _segment_raw(raw, segment_length=1., verbose=None, **kwargs):
         Segmented data.
     """
     events = make_fixed_length_events(raw, 1, duration=segment_length)
-    return Epochs(raw, events, event_id=[1], tmin=0., tmax=segment_length,
+    return Epochs(raw, events, event_id=[1], tmin=0., tmax=segment_length-1,
```

Comment:
```suggestion
    delta = 1. / raw.info['sfreq']
    return Epochs(raw, events, event_id=[1], tmin=0., tmax=segment_length - delta,
```
But there will be a PEP8 (line length > 79 char) problem so you'll need to wrap the line, too